pages/agent/v3/elastic_ci.md.erb

```diff
@@ -0,0 +1,275 @@
+
+Examples:
+
+* A `docker-builders` stack that provides always-on workers with hot Docker caches (see [Optimizing for slow Docker builds](#optimizing-for-slow-docker-builds))
+* A `pipeline-uploaders` stack with tiny, always-on instances for lightning fast `buildkite-agent pipeline upload` jobs.
+* A `deploy` stack with added credentials and permissions specifically for deployment.
+
+## Autoscaling
+
+If you configure `MinSize` < `MaxSize`, the stack automatically scales up and down based on the number of scheduled jobs.
+
+This means you can scale down to zero when idle, which means you can use larger instances for the same cost.
+
+Metrics are collected with a Lambda function, polling every minute based on the queue the stack is configured with. The autoscaler monitors only one queue.
+
+## Terminating the instance after the job is complete
+
+You can set `BuildkiteTerminateInstanceAfterJob` to `true` to force the instance to terminate after it completes a job. Setting this value to `true` tells the stack to enable `disconnect-after-job` in the `buildkite-agent.cfg` file.
+
+We strongly encourage you to find an alternative to this setting if at all possible. The turn around time for replacing these instances is currently slow (5-10 minutes depending on other stack configuration settings). If you need single use jobs, we suggest looking at our container plugins like `docker`, `docker-compose`, and `ecs`, all which can be found [here](https://buildkite.com/plugins).
+
+## Docker Registry support
+
+If you want to push or pull from registries such as [Docker Hub](https://hub.docker.com/) or [Quay](https://quay.io/) you can use the `environment` hook in your secrets bucket to export the following environment variables:
+
+* `DOCKER_LOGIN_USER="the-user-name"`
+* `DOCKER_LOGIN_PASSWORD="the-password"`
+* `DOCKER_LOGIN_SERVER=""` - optional. By default it logs in to Docker Hub
+
+Setting these performs a `docker login` before each pipeline step runs, allowing you to `docker push` to them from within your build scripts.
+
+If you use [Amazon ECR](https://aws.amazon.com/ecr/) you can set the `ECRAccessPolicy` parameter for the stack to either `readonly`, `poweruser`, or `full` depending on the [access level](http://docs.aws.amazon.com/AmazonECR/latest/userguide/ecr_managed_policies.html) you want your builds to have.
+
+You can disable this in individual pipelines by setting `AWS_ECR_LOGIN=false`.
+
+If you want to log in to an ECR server on another AWS account, you can set `AWS_ECR_LOGIN_REGISTRY_IDS="id1,id2,id3"`.
+
+The AWS ECR options are powered by an embedded version of the [ECR plugin](https://github.com/buildkite-plugins/ecr-buildkite-plugin), so if you require options that aren't listed here, you can disable the embedded version as above and call the plugin directly. See [its README](https://github.com/buildkite-plugins/ecr-buildkite-plugin) for more examples (requires Agent v3.x).
+
+## Versions
```

Comment:
Can we call this something better than Versions?  Elastic CI Stack versions? Elastic CI stack Releases?